python/lsst/ip/diffim/getTemplate.py

```diff
@@ -223,89 +232,202 @@ def run(self, coaddExposures, bbox, wcs, dataIds, physical_filter=None, **kwargs
+            is_nan = np.isnan(maskedImage.image.array)
+            # Note that modifying the patch MaskedImage in place is fine;
+            # we're throwing it away at the end anyway.
+            maskedImage.image.array[is_nan] = 0.0
+            maskedImage.variance.array[is_nan] = 0.0
+            # Reset mask, too, since these pixels don't contribute to sum.
+            maskedImage.mask.array[is_nan] = 0
+            # Cannot use `merged.maskedImage *= weight` because that operator
+            # multiplies by the weight twice; in this case `weight` are the
+            # exact values we want to scale by.
+            maskedImage.image *= weight
+            maskedImage.variance *= weight
+            merged.maskedImage[maskedImage.getBBox()] += maskedImage
+            weights[maskedImage.getBBox()] += weight
+        # Cannot use `merged.maskedImage /= weights` because that operator
+        # divides by the weight twice; in this case `weights` are the exact
+        # values we want to scale by.
+        merged.image /= weights
+        merged.variance /= weights
+        merged.mask.array |= merged.mask.getPlaneBitMask("NO_DATA") * (weights.array == 0)
+
+        return merged
+
+    def _makePsf(self, template, catalog, wcs):
+        """Return a PSF containing the PSF at each of the input regions.
+
+        Note that although this includes all the exposures from the catalog,
+        the PSF knows which part of the template the inputs came from, so when
+        evaluated at a given position it will not include inputs that never
+        went in to those pixels.
 
+        Parameters
+        ----------
+        template : `lsst.afw.image.Exposure`
+            Generated template the PSF is for.
+        catalog : `lsst.afw.table.ExposureCatalog`
+            Catalog of exposures that went into the template that contains all
+            of the input PSFs.
+        wcs : `lsst.afw.geom.SkyWcs`
+            WCS of the template, to warp the PSFs to.
+
+        Returns
+        -------
+        coaddPsf : `lsst.meas.algorithms.CoaddPsf`
+            The meta-psf constructed from all of the input catalogs.
+
+        Raises
+        ------
+        RuntimeError
+            TODO: Do we need to raise this?
+        """
         # CoaddPsf centroid not only must overlap image, but must overlap the
         # part of image with data. Use centroid of region with data.
-        boolmask = templateExposure.mask.array & templateExposure.mask.getPlaneBitMask('NO_DATA') == 0
+        boolmask = template.mask.array & template.mask.getPlaneBitMask('NO_DATA') == 0
         maskx = afwImage.makeMaskFromArray(boolmask.astype(afwImage.MaskPixel))
         centerCoord = afwGeom.SpanSet.fromMask(maskx, 1).computeCentroid()
 
         ctrl = self.config.coaddPsf.makeControl()
-        coaddPsf = CoaddPsf(tractsCatalog, finalWcs, centerCoord, ctrl.warpingKernelName, ctrl.cacheSize)
+        # TODO: how could this possibly return None?
```

Comment:
If you mean the `ctrl` object, I don't think it can.

Reply:
That's what I thought. I copied this from the old code, so I'll just remove and we can add back some other test-and-raise if we need it.